File: 01/solver1.exs
#!/usr/bin/elixir

{:ok, file} = File.read("input.txt")
lst = file |> String.trim
           |> String.split("\n")
           |> Enum.map(&String.to_integer/1)
           |> Enum.sort
# lst is now a list with sorted integers

# Find first diff w/ 2020 and yield the result:
Enum.any?(
  lst,
  fn n -> (diff = 2020 - n) &&
          Enum.member?(lst, diff) &&
          IO.puts("n: #{n}, diff: #{diff} => n * diff = #{n * diff}")
  end
)

File: 01/solver1.rb
#!/usr/bin/env ruby

ary = File.read("input.txt").split("\n").map(&:to_i).sort

ary.any? do |i|
  diff = 2020 - i
  # Bail out when done:
  done = ary.include?(diff)
  puts "i: #{i}, diff: #{diff} => i * diff = #{i * diff}" if done
  # Report the outcome to any?:
  done
end
